fg21sim/extragalactic/pointsources/base.py

```diff
@@ -0,0 +1,126 @@
+    Experiment",
+    2008,MNRAS,389,1319-1335.,
+    http://adsabs.harvard.edu/abs/2008MNRAS.389.1319W.
+"""
+import os
+import numpy as np
+from pandas import DataFrame
+
+import healpy as hp
+import astropy.units as au
+
+from .psparams import PixelParams
+
+class BasePointSource:
+    """
+    The basic class of point sources
+
+    Parameters
+    ----------
+    z: float;
+        Redshift, z ~ U(0,20)
+    dA: au.Mpc;
+        Angular diameter distance, which is calculated according to the cosmology
+        constants. In this work, it is calculated by module basic_params
+    theta: au.rad;
+        The colatitude angle in the spherical coordinate system
+    phi: au.rad;
+        The longtitude angle in the spherical coordinate system
+    area: au.sr;
+        Area of the point sources, sr = rad^2
+
+    """
+    #Init
+    def __init__(self,configs):
+        # configures
+        self.configs = configs
+        # PS_list information
+        self.columns = ['z', 'dA (Mpc)', 'Theta (deg)',
+                        'Phi (deg)', 'Area (sr)']
+        self.nCols = len(self.columns)
+        self._get_base_configs()
+
+    def _get_base_configs(self):
+        """
+        Load the configs and set the corresponding class attributes.
+        """
+        # common
+        self.nside = self.configs.getn("common/nside")
+        # save flag
+        self.save = self.configs.getn("extragalactic/pointsource/save")
+        # Output_dir
+        self.output_dir = self.configs.getn(
+                            "extragalactic/pointsource/output_dir")
+
+
+    def gen_single_ps(self):
+        """
+        Generate single point source, and return its data as a list.
+        """
+        # Redshift
+        self.z = np.random.uniform(0, 20)
+        # angular diameter distance
+        self.param = PixelParams(self.z)
+        self.dA = self.param.dA
+        # Position
+        x = np.random.uniform(0,1)
+        self.theta = np.arccos(x)/np.pi * 180 * au.deg
+        self.phi = np.random.uniform(0,np.pi*2)/np.pi * 180 * au.deg
+        # Area
+        npix = hp.nside2npix(self.nside)
+        self.area = 4*np.pi/npix * au.sr
+
+        ps_list = [self.z, self.dA.value, self.theta.value,
+                   self.phi.value, self.area.value]
+
+        return ps_list
+
+    def save_as_csv(self):
+        """
+        Generate num_ps of point sources and save them into a csv file.
+        """
+        # Init
+        ps_table = np.zeros((self.num_ps, self.nCols))
+        for x in range(self.num_ps):
+            ps_table[x, :] = self.gen_single_ps()
+
+        # Transform into Dataframe
+        ps_frame = DataFrame(ps_table, columns=self.columns,
+                             index=list(range(self.num_ps)))
+
+        # Save to csv
+        if os.path.exists(self.output_dir) == False:
+            os.mkdir(self.output_dir)
+
+        pattern = "{prefix}.csv"
+        file_name = pattern.format(prefix = self.prefix)
```

Comment:
Remove space => ``.format(prefix=self.prefix)``.

Use ``filename`` instead of ``file_name``.  Be consistent.